planner/cascades/transformation_rules.go

```diff
@@ -894,6 +894,7 @@ func (r *PushSelDownJoin) OnTransform(old *memo.ExprIter) (newExprs []*memo.Grou
 		tempCond = append(tempCond, join.OtherConditions...)
 		tempCond = append(tempCond, sel.Conditions...)
 		tempCond = expression.ExtractFiltersFromDNFs(sctx, tempCond)
+		join.SetCartesianJoin(tempCond)
```

Comment:
please also add some test cases to cover it.

Reply:
this is for sql like `select /*+ HASH_JOIN(t1) */ t1.b, t2.b from t1, t2 where t1.a = t2.a and t1.a = 10`. A new testcase add to `testIntegrationSuite.TestInlineProjection`.